### PR TITLE
Extract and write metadata on Instances

### PR DESCRIPTION
**Pull request by ngfreiter**
This addresses [CIMPL-478](https://standardhealthrecord.atlassian.net/browse/CIMPL-478) and [CIMPL-482](https://standardhealthrecord.atlassian.net/browse/CIMPL-482).

This adds extracting and writing of instances. The metadata for an instance is extracted not just based on the resource itself, but also the contents of the IG resource (if there is one). The `Usage` of an instance defaults to `#definition` if the resource is a conformance or vocabulary resource, and is `#example` otherwise. This default behavior can be overridden if there is an IG resource which indicates a `Usage` different than the default. 

This PR will have to change once #32 is merged in, but it will just be some simple refactoring to match the new style @cmoesel used there, so I think it's fine to review as is, just don't review the `FHIRProcessor.ts` code too closely, since I think that is mainly where things will change.

### CODE REVIEW

**Review by mint-thompson — APPROVED** [review]
This looks like a good foundation for handling instances. Thank you! I have one small suggestion, but it's definitely out of scope, so I'm marking this as approved.

File: src/exportable/common.ts [hunk 1]
```diff
@@ -40,13 +42,19 @@ export function metadataToFSH(
...
     resultLines.push(`Mapping: ${definition.name}`);
+  } else if (definition instanceof ExportableInstance) {
+    resultLines.push(`Instance: ${definition.name}`);
+    resultLines.push(`InstanceOf: ${definition.instanceOf}`);
   }
 
-  // Invariants do not use the Id and Title keywords
-  if (!(definition instanceof ExportableInvariant)) {
+  // Invariants and Instances don't use the "Id" keyword
+  if (!(definition instanceof ExportableInvariant || definition instanceof ExportableInstance)) {
```
[the remark this answers is not in the document]
Comment (julianxcarter) [reply src/exportable/common.ts:51]: I'd agree with what Mint said above👆🏼. That struck me as a bit odd when I was reviewing as well, I could rework as part of my PR for [CIMPL 585](https://standardhealthrecord.atlassian.net/secure/RapidBoard.jspa?rapidView=9&projectKey=INT&modal=detail&selectedIssue=CIMPL-585&sprint=74&assignee=5f7dc49c287870006a40b275) which will handle writing the fixed value rules.

**Review by julianxcarter — APPROVED** [review]
This looks really good to me too! Gonna go ahead and approve, nice work 👍🏽.

File: src/exportable/common.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ngfreiter) [reply src/exportable/common.ts:51]: Yeah I agree! I was considering making that change, and kinda wish I had just gone for it.

**Pull request merged**